### ISSUE TITLE
localstack.services.s3.s3_listener: Failed to parse LastModified

### ISSUE THREAD

**Issue by dholroyd** [localstack/localstack#2606]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

When listing an S3 bucket with more than one item, error messages are logged which dump a potentially very large amount of XML (depending on the number of objects in the bucket being listed).

For example,

```
2020-06-22T23:32:02:ERROR:localstack.services.s3.s3_listener: Failed to parse LastModified: ('Unknown string format:', '2020-06-22T23:29:35.108Z</LastModified><ETag>&#34;8d72872b23b16fc4f9c14a0dc76a8558&#34;</ET
ag><Size>796180</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358171750000</Key><LastModified>2020-06-22T23:29:40.017Z</LastModified><ETag>&#34;53d419d4a39d443a845efaae44
63b611&#34;</ETag><Size>931728</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358172200000</Key><LastModified>2020-06-22T23:29:45.016Z</LastModified><ETag>&#34;fb5c9cc82d0
3a66fe811ef4c39608dcd&#34;</ETag><Size>934360</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358172650000</Key><LastModified>2020-06-22T23:29:50.035Z</LastModified><ETag>&
#34;07f2d4652b44c1f9d961613113505ca9&#34;</ETag><Size>927780</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358173100000</Key><LastModified>2020-06-22T23:29:55.054Z</LastM
odified><ETag>&#34;0b215fb545d1561459487e04a4e9de79&#34;</ETag><Size>930412</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358173550000</Key><LastModified>2020-06-22T23:30
:00.072Z</LastModified><ETag>&#34;0d68dfa3e324e2620d2e8fb8aca32644&#34;</ETag><Size>934360</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358174000000</Key><LastModified>2
020-06-22T23:30:05.012Z</LastModified><ETag>&#34;3cd9f457157ccde76bc97e06d1616e82&#34;</ETag><Size>939624</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358174450000</Key>
<LastModified>2020-06-22T23:30:10.014Z</LastModified><ETag>&#34;c649d5ba8784af0cee3fea4acff2c07d&#34;</ETag><Size>922516</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358
174900000</Key><LastModified>2020-06-22T23:30:15.023Z</LastModified><ETag>&#34;f8325efdd76da943975fb701fbe39049&#34;</ETag><Size>935676</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/
segments/143358175350000</Key><LastModified>2020-06-22T23:30:20.041Z</LastModified><ETag>&#34;2c4630b27ea7435a86f8e1cd2530f211&#34;</ETag><Size>919884</Size><StorageClass>STANDARD</StorageClass></Contents><Conte
nts><Key>job/1/segments/143358175800000</Key><LastModified>2020-06-22T23:30:25.058Z</LastModified><ETag>&#34;8b131873c2da0d6d97f69f695fe03132&#34;</ETag><Size>939624</Size><StorageClass>STANDARD</StorageClass></
Contents><Contents><Key>job/1/segments/143358176250000</Key><LastModified>2020-06-22T23:30:30.074Z</LastModified><ETag>&#34;32abfd7695be1c24c52605634fdd1e6a&#34;</ETag><Size>931728</Size><StorageClass>STANDARD</
StorageClass></Contents><Contents><Key>job/1/segments/143358176700000</Key><LastModified>2020-06-22T23:30:35.017Z</LastModified><ETag>&#34;ac6d2ff3679e1ec6187ced080cb35d16&#34;</ETag><Size>930412</Size><StorageC
lass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358177150000</Key><LastModified>2020-06-22T23:30:40.011Z</LastModified><ETag>&#34;7eed59c427da29f42bd4a4f12fd06532&#34;</ETag><Size>917252<
/Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358177600000</Key><LastModified>2020-06-22T23:30:45.029Z</LastModified><ETag>&#34;939ca21f5cf3f72bb5a3eb738f23f48e&#34;</ETa
g><Size>930412</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358178050000</Key><LastModified>2020-06-22T23:30:50.038Z</LastModified><ETag>&#34;46cb7c6c5d6876e905e65d106d6
87df4&#34;</ETag><Size>931728</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358178500000</Key><LastModified>2020-06-22T23:30:55.060Z</LastModified><ETag>&#34;ccff83cca301
0144084ebc734d1fb4da&#34;</ETag><Size>926464</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358178950000</Key><LastModified>2020-06-22T23:31:00.082Z</LastModified><ETag>&#
34;2d9290f95f0ed630cfff7612daed4a04&#34;</ETag><Size>929096</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358179400000</Key><LastModified>2020-06-22T23:31:05.022Z</LastMo
dified><ETag>&#34;8cde7b591e5961c9220d6cfba2e950d3&#34;</ETag><Size>929096</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358180750000</Key><LastModified>2020-06-22T23:31:
20.038Z</LastModified><ETag>&#34;e54c3930f24ffa02a09ea842cdd3579e&#34;</ETag><Size>319788</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358181200000</Key><LastModified>20
20-06-22T23:31:25.072Z</LastModified><ETag>&#34;a976ffabe304f68146a6f3be41693c38&#34;</ETag><Size>936992</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358181650000</Key><
LastModified>2020-06-22T23:31:30.082Z</LastModified><ETag>&#34;0b89777db778a7aa37e1878c73173404&#34;</ETag><Size>936992</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/1433581
82100000</Key><LastModified>2020-06-22T23:31:35.010Z</LastModified><ETag>&#34;1fa96aba13d4aea70dca333095e6e634&#34;</ETag><Size>915936</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>job/1/s
egments/143358182550000</Key><LastModified>2020-06-22T23:31:40.014Z</LastModified><ETag>&#34;46a85ae0885c0dc465423152fb060415&#34;</ETag><Size>923832</Size><StorageClass>STANDARD</StorageClass></Contents><Conten
ts><Key>job/1/segments/143358183000000</Key><LastModified>2020-06-22T23:31:45.025Z</LastModified><ETag>&#34;417c183f04f9522ea628dc31e63b7f5f&#34;</ETag><Size>934360</Size><StorageClass>STANDARD</StorageClass></C
ontents><Contents><Key>job/1/segments/143358183450000</Key><LastModified>2020-06-22T23:31:50.050Z</LastModified><ETag>&#34;068f6a0f5793e8883837ac0dd676b03e&#34;</ETag><Size>931728</Size><StorageClass>STANDARD</S
torageClass></Contents><Contents><Key>job/1/segments/143358183900000</Key><LastModified>2020-06-22T23:31:55.067Z</LastModified><ETag>&#34;bdc4eb2990cff8b60e1b317d31b12cfc&#34;</ETag><Size>931728</Size><StorageCl
ass>STANDARD</StorageClass></Contents><Contents><Key>job/1/segments/143358184350000</Key><LastModified>2020-06-22T23:32:00.080Z')                                                                                  
```

## Expected behaviour

No error messages logged

## Actual behaviour

...

# Steps to reproduce

## Command used to start LocalStack

```
localstack start
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```
$ awslocal s3 mb s3://test
make_bucket: test
$ touch testfile
$ awslocal s3 cp testfile s3://test/
upload: ./testfile to s3://test/testfile
$ awslocal s3 ls s3://test
2020-06-23 10:06:58          0 testfile
$ touch testfile2
$ awslocal s3 cp testfile2 s3://test/
upload: ./testfile2 to s3://test/testfile2
$ awslocal s3 ls s3://test
```

localstack output when the **second** `awslocal s3 ls` is run...
```
2020-06-23T09:07:27:ERROR:localstack.services.s3.s3_listener: Failed to parse LastModified: ('Unknown string format:', '2020-06-23T09:06:58.279Z</LastModified><ETag>&#34;d41d8cd98f00b204e9800998ecf8427e&#34;</ETag><Size>0</Size><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>testfile2</Key><LastModified>2020-06-23T09:07:23.767Z')
```

This doesn't happen when there is only a single file in the bucket.

Given that the problematic input being quoted starts with `2020-06-23T09:06:58.279Z</LastModified>` and ends with `<LastModified>2020-06-23T09:07:23.767Z`, I presume that the problem arises from the naive use of regex in https://github.com/localstack/localstack/blob/bf209a4d36de2c533f203c0bcdcced354d6ef2b5/localstack/services/s3/s3_listener.py#L384 which is greedy-matching (the `([^<]*)`).

**Comment by dholroyd** [comment]
Actually, the code I quoted clearly can't case the problem I was guessing at there - that problem was present in a version of the code fixed months ago.  I have recently upgraded localstack but maybe I am not as up-to-date as I had assumed...

**Comment by dholroyd** [comment]
This was probably already fixed back in #1874, and I had done a `pip3 install localstack` 'update', but not a `docker pull` which has resolved this problem.  Sorry for the noise :sweat_smile:
